rest_framework/fields.py

```diff
@@ -1686,4 +1688,6 @@ def to_representation(self, obj):
         value = value_from_object(self.model_field, obj)
         if is_protected_type(value):
             return value
+        if isinstance(obj, dict):
+            return obj[self.model_field.attname]
```

Comment:
`ModelField` applies on `Model`, They are not supposed to be used for dictionaries.